pkg/athena/driver/driver.go

```diff
@@ -23,25 +25,29 @@ var (
 type Driver struct {
 	name       string
 	api        *api.API
-	connection *conn
+	asyncDB    *conn
+	connection *asyncSQLDriver.Conn
 }
 
 // Open returns a new driver.Conn using already existing settings
 func (d *Driver) Open(_ string) (driver.Conn, error) {
-	d.connection = newConnection(d.api)
 	return d.connection, nil
 }
 
 func (d *Driver) Closed() bool {
-	return d.connection == nil || d.connection.closed
+	return d.connection == nil || d.asyncDB.closed
 }
 
 func (d *Driver) OpenDB() (*sql.DB, error) {
 	return sql.Open(d.name, "")
 }
 
+func (d *Driver) GetAsyncDB() (awsds.AsyncDB, error) {
+	return d.asyncDB, nil
```

Comment:
feel free to file this under "sarah doesn't get go" but like why do we always return nil if there's no risk of error here? Is the idea that another datasource might have a situation where it needs to return an error so we always write getAsyncDB with the option to return an error?

Reply:
Yeah, that's more or less it

Reply:
haha cool thanks for bearing with my stupid questions haha